pylutron/__init__.py

```diff
@@ -93,12 +97,22 @@ def send(self, cmd):
   def _do_login_locked(self):
     """Executes the login procedure (telnet) as well as setting up some
     connection defaults like turning off the prompt, etc."""
-    self._telnet = telnetlib.Telnet(self._host)
-    self._telnet.read_until(LutronConnection.USER_PROMPT)
+    self._telnet = telnetlib.Telnet(self._host, timeout=2)  # 2 second timeout
+
+    # Ensure we know that connection goes away quickly, 5 sec
+    try:
+      to_attr = 18  # older pythons don't know about this one, so do our best
+      if hasattr(socket, 'TCP_USER_TIMEOUT'):
+        to_attr = getattr(socket, 'TCP_USER_TIMEOUT')
```

Comment:
Unfortunately, this is linux only. Does anyone know if folks actually use HA on anything other than linux? If so I'll have to sort out a more platform independent solution, cobbled together with some keepalive options.